qiskit/circuit/library/arithmetic/integer_comparator.py

```diff
@@ -0,0 +1,245 @@
+        Note that this will change the quantum registers.
+
+        Args:
+            num_state_qubits: The new number of state qubits.
+        """
+        if self._num_state_qubits is None or num_state_qubits != self._num_state_qubits:
+            self._data = None  # reset data
+            self._num_state_qubits = num_state_qubits
+
+            if num_state_qubits:
+                # set the new qubit registers
+                qr_state = QuantumRegister(self.num_state_qubits, name='state')
+                q_compare = QuantumRegister(1, name='compare')
+
+                self.qregs = [qr_state, q_compare]
+
+                if self.num_ancilla_qubits > 0:
+                    qr_ancilla = QuantumRegister(self.num_ancilla_qubits, name='ancilla')
+                    self.qregs += [qr_ancilla]
+
+    @property
+    def num_ancilla_qubits(self) -> int:
+        """The number of ancilla qubits used.
+
+        Returns:
+            The number of ancillas in the circuit.
+        """
+        return self._num_state_qubits - 1
+
+    def _get_twos_complement(self) -> List[int]:
+        """Returns the 2's complement of ``self.value`` as array.
+
+        Returns:
+             The 2's complement of ``self.value``.
+        """
+        twos_complement = pow(2, self.num_state_qubits) - int(np.ceil(self.value))
+        twos_complement = '{0:b}'.format(twos_complement).rjust(self.num_state_qubits, '0')
+        twos_complement = \
+            [1 if twos_complement[i] == '1' else 0 for i in reversed(range(len(twos_complement)))]
+        return twos_complement
+
+    @property
+    def data(self):
+        if self._data is None:
+            self._build()
+        return self._data
+
+    def _configuration_is_valid(self, raise_on_failure: bool = True) -> bool:
+        """Check if the current configuration is valid."""
+        valid = True
+
+        if self._num_state_qubits is None:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError('Number of state qubits is not set.')
+
+        if self._value is None:
+            valid = False
+            if raise_on_failure:
+                raise AttributeError('No comparison value set.')
+
+        required_num_qubits = 2 * self.num_state_qubits
+        if self.num_qubits != required_num_qubits:
+            valid = False
+            if raise_on_failure:
+                raise CircuitError('Number of qubits does not match required number of qubits.')
+
+        return valid
+
+    def _build(self) -> None:
+        """Build the comparator circuit."""
+        # set the register
+        if self._data:
+            return
+
+        _ = self._configuration_is_valid()
+
+        self._data = []
+
+        qr_state = self.qubits[:self.num_state_qubits]
+        q_compare = self.qubits[self.num_state_qubits]
+        qr_ancilla = self.qubits[self.num_state_qubits + 1:]
+
+        if self.value <= 0:  # condition always satisfied for non-positive values
```

Comment:
Hm that should not happen because the `self._configuration_is_valid()` checks whether `value` has been set, that's even part of the tests. Did you explicitly run it? What was the code snippet?

Reply:
no sorry i didn't run it, i was just reading the code. if it checks, i'm good